hc/api/views.py

```diff
@@ -424,6 +424,25 @@ def pings(request, code):
     return JsonResponse({"pings": [p.to_dict() for p in pings]})
 
 
+@cors("GET")
+@csrf_exempt
+@validate_json()
+@authorize
+def ping_body(request, code, n):
+    check = get_object_or_404(Check, code=code)
+    if check.project_id != request.project.id:
+        return HttpResponseForbidden()
+
```

Comment:
We should also check here if `n` is not too far back.  Something like (untested):

```python
profile = Profile.objects.get(user__project=request.project)
threshold = check.n_pings - profile.ping_log_limit
if n < threshold: 
    raise Http404()
```

Reply:
Done in https://github.com/healthchecks/healthchecks/pull/737/commits/4a9758054e223b2dcd5c20a4a2cdf04fdeb7d246